packages/common/src/hardhat/tasks/artifacts.js

```diff
@@ -0,0 +1,206 @@
+const path = require("path");
+const fs = require("fs");
+const uniqBy = require("lodash.uniqby");
+
+const { task, types } = require("hardhat/config");
+
+function removeFileIfExists(filename) {
+  try {
+    fs.unlinkSync(filename);
+  } catch (e) {
+    // do nothing.
+  }
+}
+
+async function getArtifactPathList(hre, relativeTo) {
+  const artifactPaths = await hre.artifacts.getArtifactPaths();
+
+  // Generate a unique list of artifacts and paths to them. Unique is necessary because there are some redundantly
+  // named contracts.
+  return uniqBy(
```

Comment:
nice use of lodash.

Reply:
Yeah, this is an interesting case. Truffle has been warning us for a while that we have some duplicate contract names. This is mostly for interfaces that we declare within contracts, like the FPLs, so they aren't contracts we really care about accessing here. So this was just to avoid generating duplicate functions.

Reply:
The existing duplicates aren't contracts we'd really use outside of solidity. They only get exported because solidity has no way to construct an interface that isn't exported.

Open to ideas for how to represent duplicates in typescript, but if we want a simple name-based export strategy, I think we have to remove duplicates to get it to compile.

Reply:
Oh! One last point on this. Duplicate contract names aren't supported by typechain in general, I don't think. They create one flat dir that contains files called `{contractName}.ts`, so it's impossible for them to represent duplicate contract names.